src/main/scala/util/Package.scala

```diff
@@ -0,0 +1,48 @@
+import Chisel._
+
+package object util {
+  implicit class UIntIsOneOf(val x: UInt) extends AnyVal {
+    def isOneOf(s: Seq[UInt]): Bool = s.map(x === _).reduce(_||_)
+  
+    def isOneOf(u1: UInt, u2: UInt*): Bool = isOneOf(u1 +: u2.toSeq)
+  }
+
+  implicit class SeqToAugmentedSeq[T <: Data](val x: Seq[T]) extends AnyVal {
+    def apply(idx: UInt): T = {
+      if (x.size == 1) {
+        x.head
+      } else {
+        val half = 1 << (log2Ceil(x.size) - 1)
+        val newIdx = idx & UInt(half - 1)
+        Mux(idx >= UInt(half), x.drop(half)(newIdx), x.take(half)(newIdx))
+      }
+    }
+
+    def asUInt(): UInt = Cat(x.map(_.asUInt).reverse)
+  }
+
```

Comment:
Importing rocket._ will now still bring them in scope, right? I'm wondering if it might have been easier just to do that into those files, but I guess we can stick with the hard rocket subdivision.

Now I'm wondering if we should have a util.DubiousImplicitConversions object that anyone can import? And that only rocket does into its own package object? Does that work?